From 15abb6b351066f5e646e0a6498bbebaaa65b1691 Mon Sep 17 00:00:00 2001
From: Jerome Brunet <jbrunet@baylibre.com>
Date: Thu, 30 Mar 2017 15:53:47 +0200
Subject: [PATCH 28/67] WIP: ARM64: dts: meson-gxbb-p20x: add sound card
 support

This commit add sound card support to the meson-gxbb-p20x
This patch is not intended to be merged upstream the architecture of
the sound card has not been determined yet. To handle the routing caps
of meson SoCs, we will probably have to make our own card.

This implementation using simple-card is provided for testing purpose only

Signed-off-by: Jerome Brunet <jbrunet@baylibre.com>
Signed-off-by: Neil Armstrong <narmstrong@baylibre.com>
---
 arch/arm64/boot/dts/amlogic/meson-gxbb-p20x.dtsi | 51 ++++++++++++++++++++++++
 1 file changed, 51 insertions(+)

diff --git a/arch/arm64/boot/dts/amlogic/meson-gxbb-p20x.dtsi b/arch/arm64/boot/dts/amlogic/meson-gxbb-p20x.dtsi
index fabc796..8926d2b 100644
--- a/arch/arm64/boot/dts/amlogic/meson-gxbb-p20x.dtsi
+++ b/arch/arm64/boot/dts/amlogic/meson-gxbb-p20x.dtsi
@@ -164,6 +164,57 @@
 		compatible = "linux,spdif-dit";
 		status = "okay";
 	};
+
+	soc {
+		sound {
+			compatible = "simple-audio-card";
+			simple-audio-card,name = "meson-gxbb-p20x";
+			simple-audio-card,aux-devs = <&amp>;
+			simple-audio-card,widgets =
+				"Line", "Analog Left Output",
+				"Line", "Analog Right Output";
+			simple-audio-card,routing =
+				"Analog Left Output", "OUTL",
+				"Analog Right Output", "OUTR",
+				"INL", "AOUTL",
+				"INR", "AOUTR";
+			status = "okay";
+
+			simple-audio-card,dai-link@0 {
+				/* RCA Output */
+				format = "i2s";
+				mclk-fs = <256>;
+				bitclock-master = <&i2s_dai>;
+				frame-master = <&i2s_dai>;
+
+				plat {
+					sound-dai = <&aiu_i2s_dma>;
+				};
+
+				cpu {
+					sound-dai = <&i2s_dai>;
+				};
+
+				codec {
+					sound-dai = <&i2s_codec>;
+				};
+			};
+
+			simple-audio-card,dai-link@1 {
+				plat {
+					sound-dai = <&aiu_spdif_dma>;
+				};
+
+				cpu {
+					sound-dai = <&spdif_dai>;
+				};
+
+				codec {
+					sound-dai = <&spdif_out>;
+				};
+			};
+		};
+	};
 };
 
 &cec_AO {
-- 
1.9.1

